Cython/Compiler/Parsing.py

```diff
@@ -3133,31 +3133,63 @@ def p_cdef_extern_block(s, pos, ctx):
 def p_c_enum_definition(s, pos, ctx):
     # s.sy == ident 'enum'
     s.next()
+
+    scoped = False
+
+    if s.sy == 'class' or (s.sy == 'IDENT' and s.systring == 'struct'):
+        scoped = True
+        s.next()
+
     if s.sy == 'IDENT':
         name = s.systring
         s.next()
         cname = p_opt_cname(s)
         if cname is None and ctx.namespace is not None:
             cname = ctx.namespace + "::" + name
     else:
-        name = None
-        cname = None
-    items = None
+        name = cname = None
+        if scoped:
+            s.error("Unnamed scoped enum not allowed")
+
+    if s.sy == '(':
```

Comment:
```suggestion
    if scoped and s.sy == '(':
```

Reply:
This seems worth another error test, BTW.

Reply:
I just pushed https://github.com/cython/cython/pull/3640/commits/bb544d5a03240287d30b07fd0c16bce3be21d2ca which I think what you are looking for, but please let me know if I misunderstood :-)